SFDigitalServices/sf-dahlia-web
# DAH-2876 feat: allow redirect after sign in

## Description

If a user tries to go to the my-account, account-settings, or my-applications and is not signed in: redirect to sign in page, show banner. Once the login happens, the user is redirected to the page they were trying to access prior.

## Jira ticket

https://sfgovdt.jira.com/browse/DAH-2876

## Checklist before requesting review

### Version Control


### Code quality


### Review instructions


### Request review



## Review Instructions
- On the review app:
- While signed out, manually enter the restricted url: /my-account?react=true
- It should try to load the page, but then redirect back to /sign-in
- You should see the banner
- Sign In to an account: test@test.com / test1234
- Once you are signed in, you should be redirected
- Try this with /my-account, /account-settings, and /my-applications

Test edge cases like signing out, triggering other banners, etc
